wireless/bluetooth/bt_hcicore.c

```diff
@@ -1622,24 +1616,35 @@ void bt_driver_unregister(FAR const struct bt_driver_s *btdev)
 
 /* TODO: rename to bt_receive? */
 
-void bt_hci_receive(FAR struct bt_buf_s *buf)
+int bt_hci_receive(FAR struct bt_driver_s *btdev,
+                   enum bt_buf_type_e type,
+                   FAR void *data, size_t len)
 {
   FAR struct bt_hci_evt_hdr_s *hdr;
+  struct bt_buf_s *buf;
   int ret;
 
-  wlinfo("buf %p len %u\n", buf, buf->len);
+  wlinfo("data %p len %zu\n", data, len);
 
   /* Critical command complete/status events use the high priority work
    * queue.
    */
 
-  if (buf->type != BT_ACL_IN)
+  buf = bt_buf_alloc(type, NULL, 0);
```

Comment:
I see a potential issue here. The buffer is typically allocated with required H4 header overhead (see code being removed in https://github.com/apache/incubator-nuttx/pull/3754/files#diff-5e9ec1dda4ad80ba37d99bf264be24a9eb50516a76bb2b0f029e59e637f2e0f4L305
This overhead is considered when consuming the bt_buf_s in netdev. Is this being accounted elsewhere?

Reply:
In the original implementation of the bt_driver, we keep the semantics of head_reserve, every implementer of the driver needs to consider the reserved data here:

https://github.com/apache/incubator-nuttx/pull/3754/files#diff-bafaf660e053e10be22390e6b4e0d76d2a60143c99508c845e21ece5999b5c2aR67